npc/re/merchants/new_insurance.txt

```diff
@@ -0,0 +1,93 @@
+		next;
+		mes "[Heart Merchant]";
+		mes "Are you curious by now? The exchange rate. <ITEM>["+.@old_item_name$+"]<INFO>" + .@old_item + "</INFO></ITEM> 5 ea. <ITEM>["+.@new_item_name$+"]<INFO>" + .@new_item + "</INFO></ITEM> 1 ea. I'll trade for one.";
+		next;
+		mes "[Heart Merchant]";
+		mes "What do you think? Isn't it a fair trade? Do you want to always worry about time? It's a must-have for adventurers who enjoy breathtaking adventures.";
+		close;
+	case 2:	
+		mes "[Heart Merchant]";
+		mes "Yeah, yeah. I thought about it. It's better to exchange what you have for a better one. Isn't it?";
+		next;
+		mes "[Heart Merchant]";
+		mes "^0000cdFor 5 "+.@old_item_name$+", you can exchange^000000 to 1 "+.@new_item_name$+".";
+		next;
+		mes "[Heart Merchant]";
+		mes "However, ^EE0000[not for sale] Life Insurance policy is not eligible for exchange^000000, so it is useless to bring it and ask for it to be replaced.";
+		next;
+		mes "[Heart Merchant]";
+		mes "Would you like to exchange it now?";
+		next;
+		if (select("Please exchange!:Don't exchange.") == 2)
+			break;
+		mes "[Heart Merchant]";
+		mes "How many do you want to exchange? ^0000cdIt is possible to exchange up to 10 "+.@new_item_name$+" at a time^000000.";
+		next;
+		input .@amount;
+		if (.@amount == 0) {
+			mes "[Heart Merchant]";
+			mes "Well? Are you not interested? It's a really nice thing to have and really good for you. But if you don't want the person to live, I can't force you.";
+			close;
+		}
+		if (.@amount > 10) {
+			mes "[Heart Merchant]";
+			mes "^0000cdYou can exchange up to 10 ea.^000000 at a time. Check the quantity again.";
+			close;
+		}
+		if (countitem(.@old_item) < (.@amount*5)) {
+			mes "[Heart Merchant]";
+			mes "Um, hey... you don't have enough life insurance policies? <ITEM>["+.@old_item_name$+"]<INFO>" + .@old_item + "</INFO></ITEM> 5 ea. <ITEM>["+.@new_item_name$+"]<INFO>" + .@new_item + "</INFO></ITEM> 1 ea. Please check the quantity.";
+			close;
+		}
+		mes "[Heart Merchant]";
+		mes "Now, with this, I exchanged the ^0000cd" + .@old_item_name$ + "^000000 that you had for the ^EE0000" + .@new_item_name$ + "^000000!! That was a good deal, right? Next time you have a "+.@old_item_name$+", come back again!";
+		delitem .@old_item,.@amount*5;
+		getitem .@new_item,.@amount;
+		close;
+	case 3:
+		break;
+	}
+	mes "[Heart Merchant]";
+	mes "This is a very good exchange! Adventurer, it's a must-have to enjoy adventures in this dangerous place!";
+	next;
+	mes "[Heart Merchant]";
+	mes "If you are interested later, be sure to come to exchange. Okay?";
+	close;
+OnTouch:
+	npctalk "You have something good. Would you like to replace your " + getitemname(12209) + " with a new one?", "", bc_self;
```

Comment:
Can you please recheck this ontouch code?
It seems like it only triggers if you have the item with you? Or some other condition maybe?
It did not trigger for us though, but we did not have a 12209 with us.

Reply:
I can't approve it now, i don't have 12209.
When I record this replays i have 12209 - 36 ea.
https://replay.divine-pride.net/Replay/9074a8f0
https://replay.divine-pride.net/Replay/c575d3b8